File: 57_InsertInterval/solution2.py
# Definition for an interval.
# class Interval:
#     def __init__(self, s=0, e=0):
#         self.start = s
#         self.end = e

class Solution(object):
    def insert(self, intervals, newInterval):
        start, end = newInterval.start, newInterval.end
        l, r = [], []
        for interval in intervals:
            if interval.end < start: l += interval,
            elif interval.start > end: r += interval,
            else:
                start = min(start, interval.start)
                end = max(end, interval.end)
        return l + [Interval(start, end)] + r
        

File: 57_InsertInterval/solution2.cpp
#include <vector>

using namespace std;

class Solution{
public:
    vector<Interval> insert(vector<Interval>& intervals, Interval newInterval){
        vector<Interval> l;
        vector<Interval> r;
        int start = newInterval.start; int end = newInterval.end;
        for(const Interval& interval : intervals){
            if(interval.end < start) l.push_back(interval);
            else if(interval.start > end) r.push_back(interval);
            else{
                start = min(start, interval.start);
                end = max(end, interval.end);
            }
        }
        vector<Interval> res(move(l));
        res.emplace_back(start, end);
        res.insert(res.end(), r.begin(), r.end());
        return res;
    }
};
